src/main/telemetry/srxl.c

```diff
@@ -172,31 +189,144 @@ void srxlFrameFlightPackCurrent(sbuf_t *dst)
     sbufWriteU16(dst, 0xffff);
     sbufWriteU16(dst, 0x7fff);            // temp B
     sbufWriteU16(dst, 0xffff);
+
+    sentAmps = amps;
+    sentCaps = caps;
+    lastTimeSentFPmAh = currentTimeUs;
+    return true;
+  }
+  return false;
+}
+
+
+#if defined (USE_SPEKTRUM_CMS_TELEMETRY) && defined (USE_CMS)
+
+// Betaflight CMS using Spektrum Tx telemetry TEXT_GEN sensor as display.
+
+#define	SPEKTRUM_SRXL_DEVICE_TEXTGEN (0x0C)     // Text Generator
+#define SPEKTRUM_SRXL_DEVICE_TEXTGEN_ROWS (9)   // Text Generator ROWS
+#define SPEKTRUM_SRXL_DEVICE_TEXTGEN_COLS (13)  // Text Generator COLS
+
+/*
+typedef struct
+{
+    UINT8		identifier;
+    UINT8		sID;		       // Secondary ID
+    UINT8		lineNumber;	       // Line number to display (0 = title, 1-8 for general, 254 = Refresh backlight, 255 = Erase all text on screen)
+    char		text[13];	       // 0-terminated text when < 13 chars
+} STRU_SPEKTRUM_SRXL_TEXTGEN;
+*/
+
+#if ( SPEKTRUM_SRXL_TEXTGEN_BUFFER_COLS > SPEKTRUM_SRXL_DEVICE_TEXTGEN_COLS )
+static char srxlTextBuff[SPEKTRUM_SRXL_TEXTGEN_BUFFER_ROWS][SPEKTRUM_SRXL_TEXTGEN_BUFFER_COLS];
+static bool lineSent[SPEKTRUM_SRXL_TEXTGEN_BUFFER_ROWS];
+#else
+static char srxlTextBuff[SPEKTRUM_SRXL_DEVICE_TEXTGEN_ROWS][SPEKTRUM_SRXL_DEVICE_TEXTGEN_COLS];
+static bool lineSent[SPEKTRUM_SRXL_DEVICE_TEXTGEN_ROWS];
+#endif
+
+//**************************************************************************
+// API Running in external client task context. E.g. in the CMS task 
+int spektrumTmTextGenPutChar(uint8_t col, uint8_t row, char c)
+{
+    if (row < SPEKTRUM_SRXL_TEXTGEN_BUFFER_ROWS && col < SPEKTRUM_SRXL_TEXTGEN_BUFFER_COLS) {
+      srxlTextBuff[row][col] = c;
+      lineSent[row] = false;
+    } else {
+      return 1;
+    }
+    return 0;
+}
+//**************************************************************************
+
+#define TEXT_GEN_KEEPALIVE_TIME_OUT 1000 // mS
```

Comment:
(1000000 us)

Reply:
Yes, 1000mS is 1000000uS.

Reply:
Using us avoids unnecessary conversion to ms / expensive division